tests/FsCodec.NewtonsoftJson.Tests/VerbatimUtf8ConverterTests.fs

```diff
@@ -24,8 +24,24 @@ type U =
     //| DT of DateTime // Have not analyzed but seems to be same issue as DTO
     | EDto of EmbeddedDateTimeOffset
     | ES of EmbeddedString
-    //| I of int // works but removed as no other useful top level values work
+    | Guid of Guid
     | N
+
+    // primitives
+    | Boolean   of bool
+    | Byte      of byte
+    | SByte     of sbyte
+    | Int16     of int16
+    | UInt16    of uint16
+    | Int32     of int32
+    | UInt32    of uint32
+    | Int64     of int64
+    | UInt64    of uint64
+    //| IntPtr  of IntPtr  // unsupported
+    //| UIntPtr of UIntPtr // unsupported
+    | Char      of char
```

Comment:
not sure about byte, sbyte and char - stuff that naturally maps to JSON seems a reasonable baseline ?

(remember Cosmos, and also things like SqlStreamStore.Postgres will impose their own constraints)

Also, related to comments on `checkIfSupported`, maybe the rules as to what's roundtripable should live in the store?

A case in point is `VerbatimUtf8Converter`: it is used by `Equinox.Cosmos`, but `Equinox.CosmosStore` (the V3 master impl), uses a different local impl (which, when we take it from https://github.com/jet/equinox/pull/197 will be based on `System.Text.Json`).

Moving the code and/or mechanisms to live with the store might be an approach to take?

A counterpoint: While EventStore is happy to store any `byte[]` blob, it also has an `IsJson` which allows one to write JS projections, and in V20+ it has a `Content-Type`.

This suggests that a reasonable codec library might be general in nature and allow one to define a profile you're targetting in the abstract, which a concrete store library can define. 

Going down that road would also mesh will with the [notion of a contract checker](https://github.com/jet/FsCodec/issues/50) which can be used to compile time check a contract adheres to rules - i.e. we might want to be able to define a convention in an integration test library that says "For this `Domain` lib, event contracts must be rountrippable on Equinox.Cosmos >=V3 and Equinox.EventStore >=V2, and all enums must be represented as strings (which implies having to add the converter attributes on the type)" 

In that world, perhaps an `FsCodec.Contracts` library might depend on TypeShape and implement all this rather than shoving everything into `FsCodec` ?

Reply:
>A case in point is `VerbatimUtf8Converter`: it is used by `Equinox.Cosmos`, but `Equinox.CosmosStore` (the V3 master impl), uses a different local impl (which, when we take it from jet/equinox#197 will be based on `System.Text.Json`).

>Moving the code and/or mechanisms to live with the store might be an approach to take?

Hm, pendulum swing: but [`Propulsion.Kafka.Codec` also uses it](https://github.com/jet/propulsion/blob/master/src/Propulsion.Kafka/Codec.fs#L35).

OK, I think that suggests that there are multiple axes of constraints:
- store-imposed: `Equinox.Cosmos` only wants JSON
- more arbitrary: `Propulsion.Kafka` is looking to roundtrip a representation of an event from a store over Kafka
- self-imposed: using `Equinox.EventStore` but we want to have all bodies be valid `JSON` so we can do JS projections
- style: our team is using F# and we have decided our rule is we use records no matter what for reasons discussed in #61
- not all roundtrips may be implemented in all libraries - i.e. some thorny thing might work in FsCodec.SystemTextJson but not in FsCodec.NewtonsoftJson

=> The rule checker may have a common impl to a degree, but we share any shareable bits by coimpiling the common code into a specific impl vs putting it into FsCodec